dts/arm/atmel/samd5x.dtsi

```diff
@@ -0,0 +1,317 @@
+		};
+
+		sercom0: sercom@40003000 {
+			compatible = "atmel,sam0-sercom";
+			reg = <0x40003000 0x40>;
+			interrupts = <46 0>, <47 0>, <48 0>, <49 0>;
+			status = "disabled";
+			label = "SERCOM0";
+		};
+
+		sercom1: sercom@40003400 {
+			compatible = "atmel,sam0-sercom";
+			reg = <0x40003400 0x40>;
+			interrupts = <50 0>, <51 0>, <52 0>, <53 0>;
+			status = "disabled";
+			label = "SERCOM1";
+		};
+
+		sercom2: sercom@41012000 {
+			compatible = "atmel,sam0-sercom";
+			reg = <0x41012000 0x40>;
+			interrupts = <54 0>, <55 0>, <56 0>, <57 0>;
+			status = "disabled";
+			label = "SERCOM2";
+		};
+
+		sercom3: sercom@41014000 {
+			compatible = "atmel,sam0-sercom";
+			reg = <0x41014000 0x40>;
+			interrupts = <58 0>, <59 0>, <60 0>, <61 0>;
+			status = "disabled";
+			label = "SERCOM3";
+		};
+
+		sercom4: sercom@43000000 {
+			compatible = "atmel,sam0-sercom";
+			reg = <0x43000000 0x40>;
+			interrupts = <62 0>, <63 0>, <64 0>, <65 0>;
+			status = "disabled";
+			label = "SERCOM4";
+		};
+
+		sercom5: sercom@43000400 {
+			compatible = "atmel,sam0-sercom";
+			reg = <0x43000400 0x40>;
+			interrupts = <66 0>, <67 0>, <68 0>, <69 0>;
+			status = "disabled";
+			label = "SERCOM5";
+		};
+
+		sercom6: sercom@43000800 {
+			compatible = "atmel,sam0-sercom";
+			reg = <0x43000800 0x40>;
+			interrupts = <70 0>, <71 0>, <72 0>, <73 0>;
+			status = "disabled";
+			label = "SERCOM6";
+		};
+
+		sercom7: sercom@43000C00 {
+			compatible = "atmel,sam0-sercom";
+			reg = <0x43000C00 0x40>;
+			interrupts = <74 0>, <75 0>, <76 0>, <77 0>;
+			status = "disabled";
+			label = "SERCOM7";
+		};
+
+		porta: gpio@41008000 {
+			compatible = "atmel,sam0-gpio";
+			reg = <0x41008000 0x80>;
+			label = "PORTA";
+			gpio-controller;
+			#gpio-cells = <2>;
+		};
+
+		portb: gpio@41008080 {
+			compatible = "atmel,sam0-gpio";
+			reg = <0x41008080 0x80>;
+			label = "PORTB";
+			gpio-controller;
+			#gpio-cells = <2>;
+		};
+
+		portc: gpio@41008100 {
+			compatible = "atmel,sam0-gpio";
+			reg = <0x41008100 0x80>;
+			label = "PORTC";
+			gpio-controller;
+			#gpio-cells = <2>;
+		};
+
+		portd: gpio@41008180 {
+			compatible = "atmel,sam0-gpio";
+			reg = <0x41008180 0x80>;
+			label = "PORTD";
+			gpio-controller;
+			#gpio-cells = <2>;
+		};
+
+		usb0: usb@41000000 {
+			compatible = "atmel,sam0-usb";
+			status = "disabled";
+			reg = <0x41000000 0x1000>;
+			interrupts = <80 0>, <81 0>, <82 0>, <83 0>;
+			num-bidir-endpoints = <8>;
+			label = "USB0";
+		};
+
+		rng: rng@42002800 {
+			compatible = "atmel,sam0-trng";
+			reg = <0x42002800 0x1e> ;
+			interrupts = <131 0>;
+			label = "ENTROPY_0";
+		};
+
+		rtc: rtc@40002400 {
+			compatible = "atmel,sam0-rtc";
+			reg = <0x40002400 0x1C>;
+			interrupts = <11 0>;
+			status = "disabled";
+			label = "RTC";
+		};
+
+		adc0: adc@43001C00 {
+			compatible = "atmel,sam0-adc";
+			reg = <0x43001C00 0x4A>;
+			interrupts = <118 0>, <119 0>;
+			label = "ADC_0";
+
+			/* FIXME: is this stil true?
```

Comment:
This looks like a bit of an odd case: the SAMD5x manual says fGCLK_ADC max is 100 MHz (table 54-8, in section 54.6, p 2020), while the F_ADC is 16 MHz (table 54-24, section 54.10.4, p 2031).  So though you can run it faster (i.e. 16 MHz after division for the rated 1 MSPS), clocking it off the full 120 MHz GCLK0 is technically out of spec, as I read it.

Reply:
Thank you! I hadn't really looked into this yet, just wanted something to compile your ADC driver with 😄 

Since currently I only configure 2 clocks, the main clock (120MHz, but can be set in KConfig) and a 32kHz clock, I suppose I should default to the latter here if I don't want to introduce a separate clock just for ADC.

I suppose this will be much cleaner with #15585